src/components/custom-field-input-text/custom-field-input-text.jsx

```diff
@@ -33,8 +33,25 @@ export default function CustomFieldInputText(props) {
     }
   });
 
-  const icon = () => {
-    if (isInvalid(props.error, props.readOnly)) {
+  const numIcons = () => {
```

Comment:
Wow this really clever! I'm worried that it could be too clever for its own good. What if we extracted this out to a hook? Something like

```jsx
import { useState, useEffect } from 'react';

export default function useIconTally(isInvalid, customFieldProps) {
  const cssVarName = `--numIcon-${customFieldProps.id}`;
  const [numIcons, setNumIcons] = useState(0);
  const [iconStyles, setIconStyles] = useState({ [cssVarName]: numIcons })

  useEffect(() => {
    let num = 0;

    if (isInvalid) {
      num += 1;
    }

    if (customFieldProps.icon) {
      num += 1;
    }

    if (customFieldProps.clear) {
      num += 1;
    }

    setNumIcons(num);
    setIconStyles({ [cssVarName]: num });
  });

  return [numIcons, iconStyles];
}
```

You could test it with `@testing-library/render-hook`, and then we have an extra hook to use internally across each of the custom fields as needed.

Reply:
Also, if we use #145's refactor of extracting an abstract custom field, we could include this as a hook on that, and then every custom field we use would receive this behavior.